### PR TITLE
Dead plugins

### PR DESCRIPTION
**Pull request by PatrikLundell**
Addresses parts of df-structures issue 247.

### CODE REVIEW

**Comment by lethosor** [comment]
Link: DFHack/df-structures#247

(You can also just paste the full link to the issue and it'll display like that.)

File: plugins/cursecheck.cpp [hunk 1]
```diff
@@ -160,8 +160,9 @@ command_result cursecheck (color_ostream &out, vector <string> & parameters)
     {
         df::unit * unit = world->units.all[i];
 
+        // filter out all "living" units that are currently removed from play
         // don't spam all completely dead creatures if not explicitly wanted
-        if(unit->flags1.bits.dead && ignoreDead)
+        if((unit->flags1.bits.dead && !unit->flags2.bits.killed) || (unit->flags2.bits.killed && ignoreDead))
```
Comment (lethosor) [inline plugins/cursecheck.cpp]: It would be clearer to use library functions from #1297 here once that's merged.

Comment (PatrikLundell) [reply plugins/cursecheck.cpp]: True, but I can't do that before then, at least not if I wan't it to compile in my end.

**Comment by PatrikLundell** [comment]
As can be seen, I've switched to Units functions. I've compiled it by first copying the Units.h and Units.cpp over from the dead_units branch without adding/committing them.

**Comment by PatrikLundell** [comment]
I did get one of the commit comments wrong (guess which one).
These changes should mean the files won't have to be changed when the "dead" flag is renamed, and all of the files did make use of other Units functions anyway.
Obviously, it still won't compile, though.

**Pull request merged**